File: awesome_map_mobile/lib/models/event/event.dart
import 'package:awesome_map_mobile/models/base/category.dart';
import 'package:awesome_map_mobile/models/comment/comment.dart';
import 'package:awesome_map_mobile/models/files/serverFile.dart';
import 'package:json_annotation/json_annotation.dart';
part 'event.g.dart';

@JsonSerializable()
class Event {
  @JsonKey(required: true)
  String id;
  @JsonKey(required: true)
  double latitude;
  @JsonKey(required: true)
  double longitude;
  @JsonKey(required: true)
  String title;
  @JsonKey(required: true)
  List<Category> eventTypes;
  String description;
  String placeDescription;
  DateTime startDate;
  DateTime createdDate;
  Duration duration;
  int peopleCount;
  bool isClosed;
  int subscribersCount;
  List<ServerFile> files;
  List<Comment> comments;

  Event(
      this.id,
      this.latitude,
      this.longitude,
      this.title,
      this.eventTypes,
      this.description,
      this.placeDescription,
      this.startDate,
      this.createdDate,
      this.duration,
      this.peopleCount,
      this.isClosed,
      this.subscribersCount,
      this.files,
      this.comments);

  factory Event.empty() {
    return Event(
        "00000000-0000-0000-0000-000000000000",
        0,
        0,
        "",
        new List<Category>(),
        "",
        "",
        DateTime.now(),
        DateTime.now(),
        null,
        0,
        false,
        0,
        new List<ServerFile>(),
        new List<Comment>());
  }

  factory Event.fromJson(Map<String, dynamic> json) => _$EventFromJson(json);

  Map<String, dynamic> toJson() => _$EventToJson(this);

  addCategory(Category item) {
    this.eventTypes.add(item);
  }

  removeCategory(String guid) {
    this.eventTypes = this.eventTypes.where((item) => item.id != guid).toList();
  }
}

File: awesome_map_server/DataBaseModels/Event/Event.cs
using DataBaseModels.Models.Base;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace DataBaseModels.Models {
    public class Event : Ticket {
        public Guid Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public string PlaceDescription { get; set; }
        public DateTime StartDate { get; set; }
        public TimeSpan Duration { get; set; }
        public DateTime EndDate { get { return StartDate + Duration; } }
        public int? PeopleCount { get; set; }

        public string OwnerId { get; set; }
        public virtual ApplicationUser Owner { get; set; }

        public virtual List<EventTypeEvent> EventTypeEvents { get; set; }

        public virtual List<EventUser> Subscribers { get; set; }
        public bool IsClosed { get; set; }
        public virtual List<Comment> Comments { get; set; }
        public virtual List<ServerFile> Files { get; set; }


        public Event() {
            EventTypeEvents = new List<EventTypeEvent>();
            Subscribers = new List<EventUser>();
            Comments = new List<Comment>();
            Files = new List<ServerFile>();
        }
    }
}
